### DISCUSSION TITLE
ERROR ReferenceError: window is not defined angular version 18

### DISCUSSION THREAD

**Discussion by arunkumar12712** [ghiscoding/Angular-Slickgrid#1474] [General]
<img width="804" alt="image" src="https://github.com/user-attachments/assets/114aca4d-00f0-4347-9c40-47038de9bd47">

**Comment by ghiscoding** [comment]
I'm assuming that it's you that asked for the exact same question in Stack Overflow [window is not define when are angular-slickgrid, Angular 18](https://stackoverflow.com/questions/78990039/window-is-not-define-when-are-angular-slickgrid-angular-18). The answer is the same as what was already mentioned in there

> If you don't need SSR, you should disable it.

The thing is that Angular-Slickgrid requires JavaScript to run and a lot of the code is dynamic so it does require the `window` object which is not supported in SSR from what I could find (I'm not entirely sure since I have never used SSR yet but from what I read you should only use it for something that doesn't heavily require JavaScript). I don't have further info but I'm assuming that you will have to disable SSR for at least Angular-Slickgrid.

**Reply by arunkumar12712** [reply]
thanks problem solved
